thunder/core/rematerialization.py

```diff
@@ -125,7 +127,7 @@ def apply_rematerialization_for_producer(
     )
     all_produced_vars = tuple(chain.from_iterable((y for y in x.flat_outs) for x in producer.subsymbols))
     # Choose the new producer's output from all the produced variables.
-    new_producer_output = tuple(x for x in all_produced_vars if x.name in new_producer_output_names)
+    new_producer_output = tuple(x for x in all_produced_vars if x is not None and x.name in new_producer_output_names)
```

Comment:
We can revert this change.
```suggestion
    new_producer_output = tuple(x for x in all_produced_vars if x.name in new_producer_output_names)
```